lib/chartUtils.ts

```
@@ -0,0 +1,93 @@
+
+export function calculateChartDomain(
+  data: Record<string, unknown>[],
+  yKeys: string[]
+): ChartDomainData {
+  if (data.length === 0) return { yDomain: [0, 100], yTickValues: [0, 25, 50, 75, 100] };
+
+  const yValues = data.flatMap(item =>
+    yKeys
+      .map(key => {
+        const value = item[key];
+        return typeof value === 'number' ? value : parseFloat(String(value) || '0');
```

Biome (1.9.4)

[error] Use Number.parseFloat instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseFloat instead.


(lint/style/useNumberNamespace)